src/main/java/net/snowflake/client/core/arrow/ThreeFieldStructToTimestampTZConverter.java

```diff
@@ -37,6 +37,11 @@ public ThreeFieldStructToTimestampTZConverter(
     timeZoneIndices = structVector.getChild(FIELD_NAME_TIME_ZONE_INDEX, IntVector.class);
   }
 
+  @Override
+  public boolean isNull(int index) {
```

Comment:
Without this, the parent `isNull` method is called, but that return value doesn't match `epochs.isNull`.

Reply:
It's interesting the parent `isNull` has an inconsistent behavior...

Reply:
Thanks for fixing this and my fault to not add this in all TZ converters (I checked LTZ and NTZ should be fine)! I double checked the serverside code, we did set null for the parent and all fields for a null value. It could be a bug in the Arrow Java Library that didn't handle this correctly because seems we didn't have the issue in Python.